simple_graph.py

```diff
@@ -93,3 +95,158 @@ def adjacent(self, n1, n2):
                 return True
 
         return False
+
+    def depth_first_traversal(self, start, discovered=None):
+        """Perform a full depth-first traversal of the graph beginning at
+        start. Return the full visited path when traversal is complete.
+        """
+        if discovered is None:
+            discovered = []
+
+        discovered.append(start)
+        for neighbor in self.neighbors(start):
+            if neighbor not in discovered:
+                self.depth_first_traversal(neighbor, discovered=discovered)
+        return discovered
+
+    def breadth_first_traversal(self, start):
+        """Perform a full breadth-first traversal of the graph, beginning at
+        start. Return the full visited path when traversal is complete.
+        """
+        result = []
+        q = Queue()
```

Comment:
What's going to happen when you put a breakpoint into this method to debug it and you need to evaluate the value of `q`?  How about the value of `n` below? 

Don't use one-letter variable names.